src/main/java/org/commonjava/cdi/util/weft/PoolWeftExecutorService.java

```diff
@@ -131,79 +135,144 @@ public boolean awaitTermination( long l, TimeUnit timeUnit )
         return delegate.awaitTermination( l, timeUnit );
     }
 
+    private boolean isOverloaded()
+    {
+        return loadSensitive && !isHealthy();
+    }
+    
     @Override
     public <T> Future<T> submit( Callable<T> callable )
     {
+        if ( isOverloaded() )
+        {
+            throw new PoolOverloadException( getName(), getLoadFactor(), getCurrentLoad(), getThreadCount() );
+        }
```

Comment:
How about move the throw clause into the function too ?

Reply:
Sounds good.